llama/patches/0029-fix-vulkan-PCI-ID-and-ID-handling.patch

```diff
@@ -0,0 +1,236 @@
+Subject: [PATCH] fix vulkan PCI ID and ID handling
+
+---
+ ggml/include/ggml-backend.h          |  3 ---
+ ggml/src/ggml-cuda/ggml-cuda.cu      | 15 +++------------
+ ggml/src/ggml-impl.h                 |  2 +-
+ ggml/src/ggml-vulkan/ggml-vulkan.cpp | 27 +++++++++++++--------------
+ ggml/src/mem_hip.cpp                 | 20 ++++++++++++--------
+ 5 files changed, 29 insertions(+), 38 deletions(-)
+
+diff --git a/ggml/include/ggml-backend.h b/ggml/include/ggml-backend.h
+index 094fc3c82..809835243 100644
+--- a/ggml/include/ggml-backend.h
++++ b/ggml/include/ggml-backend.h
+@@ -174,9 +174,6 @@ extern "C" {
+         int compute_major;
+         int compute_minor;
+         int integrated;
+-        int pci_bus_id;
+-        int pci_device_id;
+-        int pci_domain_id;
+         const char *library;
+         // number with which the devices are accessed (Vulkan)
+         const char *numeric_id;
+diff --git a/ggml/src/ggml-cuda/ggml-cuda.cu b/ggml/src/ggml-cuda/ggml-cuda.cu
+index 816597d2f..b075a18be 100644
+--- a/ggml/src/ggml-cuda/ggml-cuda.cu
++++ b/ggml/src/ggml-cuda/ggml-cuda.cu
+@@ -3504,9 +3504,6 @@ struct ggml_backend_cuda_device_context {
+     int driver_major;
+     int driver_minor;
+     int integrated;
+-    int pciBusID;
+-    int pciDeviceID;
+-    int pciDomainID;
+ };
+ 
+ static const char * ggml_backend_cuda_device_get_name(ggml_backend_dev_t dev) {
+@@ -3530,9 +3527,9 @@ static void ggml_backend_cuda_device_get_memory(ggml_backend_dev_t dev, size_t *
+ 
+ #if defined(GGML_USE_HIP)
+     if (ggml_hip_mgmt_init() == 0) {
+-        int status = ggml_hip_get_device_memory(ctx->pciBusID, ctx->pciDeviceID, free, total);
++        int status = ggml_hip_get_device_memory(ctx->pci_bus_id.c_str(), free, total);
+         if (status == 0) {
+-            GGML_LOG_DEBUG("%s utilizing ADLX memory reporting free: %zu total: %zu\n", __func__, *free, *total);
++            GGML_LOG_DEBUG("%s device %s utilizing ADLX memory reporting free: %zu total: %zu\n", __func__, ctx->pci_bus_id.c_str(), *free, *total);
+             ggml_hip_mgmt_release();
+             return;
+         }
+@@ -3542,7 +3539,7 @@ static void ggml_backend_cuda_device_get_memory(ggml_backend_dev_t dev, size_t *
+     if (ggml_nvml_init() == 0) {
+         int status = ggml_nvml_get_device_memory(ctx->id.c_str(), free, total);
+         if (status == 0) {
+-            GGML_LOG_DEBUG("%s utilizing NVML memory reporting free: %zu total: %zu\n", __func__, *free, *total);
++            GGML_LOG_DEBUG("%s device %s utilizing NVML memory reporting free: %zu total: %zu\n", __func__, ctx->id.c_str(), *free, *total);
+             ggml_nvml_release();
+             return;
+         }
+@@ -3582,9 +3579,6 @@ static void ggml_backend_cuda_device_get_props(ggml_backend_dev_t dev, ggml_back
+     props->driver_major = ctx->driver_major;
+     props->driver_minor = ctx->driver_minor;
+     props->integrated = ctx->integrated;
+-    props->pci_bus_id = ctx->pciBusID;
```

Comment:
Could this not be removed in the 0026-GPU ... patch so avoiding this and then removing it again